bundles/persistence/org.openhab.persistence.jdbc/java/org/openhab/persistence/jdbc/internal/JdbcConfiguration.java

```diff
@@ -0,0 +1,321 @@
+		serviceName = parsedURL.getProperty("dbShortcut"); // derby, h2, hsqldb, mariadb, mysql, postgresql, sqlite
+		logger.debug("JDBC::updateConfig: found serviceName = '{}'", serviceName);
+		if (StringUtils.isBlank(serviceName)) {
+			logger.error(
+					"JDBC::updateConfig: url Required database url like 'jdbc:<service>:<host>[:<port>;<attributes>]' - please configure the jdbc:url parameter in openhab.cfg");
+		}
+
+		// DB Class		
+		String ddp = DB_DAO_PACKAGE + serviceName.toUpperCase().charAt(0) + serviceName.toLowerCase().substring(1)+ "DAO";
+		
+		logger.debug("JDBC::updateConfig: Init Data Access Object Class: '{}'", ddp);
+		try {
+			
+			dBDAO = (JdbcBaseDAO) Class.forName(ddp).newInstance();
+			//dBDAO.databaseProps.setProperty("jdbcUrl", url);
+			//dBDAO.databaseProps.setProperty("dataSource.url", url);
+			
+			logger.debug("JDBC::updateConfig: dBDAO ClassName={}", dBDAO.getClass().getName());
+		} catch (InstantiationException e) {
+			logger.error("JDBC::updateConfig: InstantiationException: {}", e.getMessage());
+		} catch (IllegalAccessException e) {
+			logger.error("JDBC::updateConfig: IllegalAccessException: {}", e.getMessage());
+		} catch (ClassNotFoundException e) {
+			logger.warn("JDBC::updateConfig: no Configuration for serviceName '{}' found. ClassNotFoundException: {}", serviceName,
+					e.getMessage());
+			logger.debug("JDBC::updateConfig: using default Database Configuration: JdbcBaseDAO !!");
+			dBDAO = new JdbcBaseDAO();
+			logger.debug("JDBC::updateConfig: dBConfig done");
+		}
+		
+		@SuppressWarnings("unchecked")
+		Enumeration<String> keys = new IteratorEnumeration(configuration.keySet().iterator());
+
+		while (keys.hasMoreElements()) {
+			String key = keys.nextElement();
+
+			Matcher matcher = EXTRACT_CONFIG_PATTERN.matcher(key);
+
+			if (!matcher.matches()) {
+				continue;
+			}
+
+			matcher.reset();
+			matcher.find();
+
+			if (!matcher.group(1).equals("sqltype"))
+				continue;
+
+			String itemType = matcher.group(2).toUpperCase() + "ITEM";
+			String value = (String) configuration.get(key);
+
+			logger.debug("JDBC::updateConfig: set sqlTypes: itemType={} value={}", itemType, value);
+			dBDAO.sqlTypes.put(itemType, value);
+		}
+
+		String user = (String) configuration.get("user");
+		logger.debug("JDBC::updateConfig:  user={}", user);
+		if (StringUtils.isBlank(user)) {
+			logger.error(
+					"JDBC::updateConfig: SQL user is missing - please configure the jdbc:user parameter in openhab.cfg");
+		}else{
+			dBDAO.databaseProps.setProperty("dataSource.user", user);
+		}
+
+		String password = (String) configuration.get("password");
+		if (StringUtils.isBlank(password)) {
+			logger.error(
+					"JDBC::updateConfig: SQL password is missing. Attempting to connect without password. To specify a password configure the jdbc:password parameter in openhab.cfg.");
+		}else{
+			logger.debug("JDBC::updateConfig:  password={}{}...", password.subSequence(0, 3), password.length() - 3);
```

Comment:
nitpick: IndexOutOfBoundsException would happen if the password is shorter than 4 characters.